src/sentry/tasks/assemble.py

```diff
@@ -53,6 +53,82 @@ class AssembleTask:
     ARTIFACT_BUNDLE = "organization.artifact_bundle"  # Artifact bundle upload
 
 
+AssembleResult = collections.namedtuple("AssembleResult", ["bundle", "temp_file"])
+
+
+def assemble_file(
+    task, org_or_project, name, checksum, chunks, file_type
+) -> Optional[AssembleResult]:
+    """
+    Verifies and assembles a file model from chunks.
+
+    This downloads all chunks from blob store to verify their integrity and
+    associates them with a created file model. Additionally, it assembles the
+    full file in a temporary location and verifies the complete content hash.
+
+    Returns a tuple ``(File, TempFile)`` on success, or ``None`` on error.
+    """
+    from sentry.models import AssembleChecksumMismatch, File, FileBlob, Project
+
+    if isinstance(org_or_project, Project):
+        organization = org_or_project.organization
+    else:
+        organization = org_or_project
+
+    # Load all FileBlobs from db since we can be sure here we already own all
+    # chunks need to build the file
+    file_blobs = FileBlob.objects.filter(checksum__in=chunks).values_list("id", "checksum", "size")
+
+    # Reject all files that exceed the maximum allowed size for this
+    # organization. This value cannot be
```

Comment:
cannot be what?

Reply:
Keep in mind that lots of code has been moved for this PR, thus old weird code will show in the diff. Will fix it.